### PR TITLE
Handle `SSH_MSG_CHANNEL_EXTENDED_DATA` msg type

### PR DESCRIPTION
**Pull request by mustard-mh**
See [RFC](https://www.ietf.org/rfc/rfc4254.txt#:~:text=be%20passed%20with-,SSH_MSG_CHANNEL_EXTENDED_DATA,-messages%2C%20where%20a), search SSH_MSG_CHANNEL_EXTENDED_DATA

Fix https://github.com/microsoft/dev-tunnels-ssh/issues/68

### CODE REVIEW

File: src/ts/ssh/sshChannel.ts [hunk 1]
```diff
@@ -271,6 +276,61 @@ export class SshChannel implements Disposable {
 		}
 	}
 
+	public async sendStderr(data: Buffer, cancellation?: CancellationToken): Promise<void> {
```
Comment (IlyaBiryukov) [inline src/ts/ssh/sshChannel.ts:279]: How about making it flexible: name it `sendExtendedData()` and make `dataTypeCode` an argument?

**Review by IlyaBiryukov — DISMISSED** [review]

File: src/ts/ssh/sshChannel.ts (hunk 1 above)
Comment (IlyaBiryukov) [reply src/ts/ssh/sshChannel.ts:279]: `sendStderr` may then just call it with `1` data type code.

Comment (mustard-mh) [reply src/ts/ssh/sshChannel.ts]: @IlyaBiryukov , I pushed changes (see [diff](https://github.com/microsoft/dev-tunnels-ssh/compare/eb479a142f45dc8c7e6a7ace4da2b687280498ae...bede4e0942645b087614102ad703861485d43120?expand=1)) to address this comment

File: src/ts/ssh/sshChannel.ts [hunk 2]
```diff
@@ -388,6 +449,11 @@ export class SshChannel implements Disposable {
 		this.dataReceivedEmitter.fire(data);
 	}
 
+	public handleExtendedDataReceived(data: SshExtendedDataEventArgs): void {
+		// this.metrics.addBytesReceived(data.length);
```
Comment (jasongin) [inline src/ts/ssh/sshChannel.ts:453]: Extended data should be included in channel metrics.

File: src/ts/ssh/sshChannel.ts [hunk 3]
```diff
@@ -271,6 +277,61 @@ export class SshChannel implements Disposable {
 		}
 	}
 
+	public async sendExtendedData(dataTypeCode: SshExtendedDataType, data: Buffer, cancellation?: CancellationToken): Promise<void> {
```
Comment (jasongin) [inline src/ts/ssh/sshChannel.ts:280]: Would it be possible to refactor to eliminate some of the duplication of code between this method and `send()` above?

**Review by jasongin — COMMENTED** [review]
Thanks for the PR! It would be great to implement this also in the C# lib. But that can be done separately.

File: src/ts/ssh/sshChannel.ts (hunk 1 above)
Comment (IlyaBiryukov) [reply src/ts/ssh/sshChannel.ts]: Thanks! The change looks good for me. I'll defer to Jason to sign off, he has more comments.

File: src/ts/ssh/sshChannel.ts (hunk 3 above)
Comment (mustard-mh) [reply src/ts/ssh/sshChannel.ts:230]: @jasongin  Addressed, but don't know if you like `sendCommon` function or not 🙏

**Comment by jasongin** [comment]
Published to npm as version `3.11.24`.

**Pull request merged**